### PR TITLE
new repo/developer activity & influence trend charts

### PR DESCRIPTION
**Pull request by tyn1998**
This PR is initiated by #297, and any updates will be put here.

### CODE REVIEW

**Comment by tyn1998** [comment]
The first thing we should **decide on** is where to insert the new charts on both repo home page and user profile page.

For repo home page, I have two options now:

1. above code directory
image

2. below Languages
image

For user profile page:
image

What do you guys think? @frank-zsy @zhuxiangning @heming6666 @LiuChangFreeman @will-ww

**Comment by frank-zsy** [comment]
I think the 2nd option of repo and user profile are fine as they are identical in layout logic and more elegant. I think the space above code is too wide and too aggressive, fix width is good enough for me.

**Comment by tyn1998** [comment]
Agree~

**Comment by tyn1998** [comment]
Successfully anchored:
image

**Comment by tyn1998** [comment]
A demo (from vscode repo page): 

image

**Comment by tyn1998** [comment]
This is @frank-zsy 's trends of Activity & Influence:
image

**Comment by tyn1998** [comment]
Almost finished. Guys you can put your comment below and I will make it better.

**Comment by tyn1998** [comment]
@frank-zsy, why 404 NOT FOUND "https://hypertrons-oss.x-lab.info/activity_influence/actors/x/xiaoya-Esther.json"? Is this data not computed?

**Comment by tyn1998** [comment]
Do you think we should add i18n suuport for this charts? @frank-zsy

**Comment by frank-zsy** [comment]
> Do you think we should add i18n suuport for this charts? @frank-zsy

Yes, I suppose i18n is preferred for all components in hypercrx.

**Comment by tyn1998** [comment]
ok~

**Comment by frank-zsy** [comment]
And can split activity and influence into tow different y axis since the value range is different?

**Comment by tyn1998** [comment]
Things can also be done with [multiple-y-axis charts](https://echarts.apache.org/examples/zh/editor.html?c=multiple-y-axis), and then we will have left-y and right-y to measure activity and influence separately. Which way do you prefer?
image

**Comment by frank-zsy** [comment]
I think activity on left-y axis and influence on right-y axis is just fine since activity is easier to understand.

**Comment by tyn1998** [comment]
got~ I will make and update some demos later today.

**Comment by tyn1998** [comment]
How about this?

**Comment by frank-zsy** [comment]
I think it is good enough to be a release~

**Comment by tyn1998** [comment]
Then I will add i18n support and update this PR.

**Comment by tyn1998** [comment]
en:
image
zh:
image

**Comment by tyn1998** [comment]
@frank-zsy @LiuChangFreeman @heming6666 Guys, this PR is ready to be reviewed!

File: src/api/developer.ts [hunk 1]
```diff
@@ -17,3 +17,9 @@ export const getParticipatedProjects = async (developer: string) => {
     (await request(`/actor/${developer}_top.json`))
   );
 };
+
+export const getDeveloperActiInfl = async (developer: string) => {
+  return await request(
+    `/activity_influence/actors/${developer[0]}/${developer}.json`
```
Comment (frank-zsy) [inline src/api/developer.ts:23]: I assume you didn't change the OSS file path to `oss.x-lab.info`?

File: src/api/developer.ts [hunk 2]
```diff
@@ -7,13 +8,19 @@ import {
...
+    (await request(`${HYPERTRONS_OSS_XLAB_ENDPOINT}/actor/${developer}.json`))
   );
 };
 
 export const getParticipatedProjects = async (developer: string) => {
   return (
     mockSuccessRes(participatedProjectsData) ||
-    (await request(`/actor/${developer}_top.json`))
+    (await request(
+      `${HYPERTRONS_OSS_XLAB_ENDPOINT}/actor/${developer}_top.json`
```
Comment (frank-zsy) [inline src/api/developer.ts:19]: What is developer_top.json? I don't think the file is online right now

Comment (tyn1998) [reply src/api/developer.ts:19]: developer_top.json is the data source of our graph charts and they are stored in hypertrons-oss.xlab.info. There are two data endpoints now.

Comment (frank-zsy) [reply src/api/developer.ts:19]: Yes, but I think you use the same endpoints here or the `HYPERTRONS_OSS_XLAB_ENDPOINT` is different ?

Comment (frank-zsy) [reply src/api/developer.ts:19]: My fault, misunderstand here.

**Review by frank-zsy — APPROVED** [review]

**Comment by frank-zsy** [comment]
Really looking forward to this feature, is there any other problems about this PR?

**Comment by tyn1998** [comment]
Yes, there is. I just found a bug. I will fix this.

**Comment by tyn1998** [comment]
light & dark themes will also be supported. I'm working on this.

**Comment by tyn1998** [comment]
I examined echarts options today and made interactions with the charts better, especially in terms of data zooming.

It looks like below now(in dark mode):

**Comment by tyn1998** [comment]
I think the PR is prepared to be merged :) @frank-zsy

**Pull request merged**